### PR TITLE
Add Horse hotkey feature

### PR DESCRIPTION
**Pull request by ryanzhoudev**
Adds horse hotkey feature

Also fixes that 1.12.2 bug where pressing the key would open the quest book if you're holding it

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/MountHorseHotkeyFeature.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            ItemStack stack = player.getInventory().getItem(i);
+            if (stack.getItem() == Items.SADDLE
+                    && stack.getDisplayName().getString().contains("Horse")) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    private static void postHorseErrorMessage(MountHorseStatus status) {
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/features/MountHorseHotkeyFeature.java:167]: imo should be a enum parameter

File: common/src/main/java/com/wynntils/features/MountHorseHotkeyFeature.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        }
+        return null;
+    }
+
+    private static int findHorseInInventory() {
+        Player player = McUtils.player();
+        for (int i = 0; i <= 44; i++) {
+            ItemStack stack = player.getInventory().getItem(i);
+            if (stack.getItem() == Items.SADDLE
+                    && stack.getDisplayName().getString().contains("Horse")) {
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/features/MountHorseHotkeyFeature.java]: should be in `WynnItemMatchers`

File: common/src/main/java/com/wynntils/features/MountHorseHotkeyFeature.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        String playerName = McUtils.player().getName().getString();
+        String defaultName = "§f" + playerName + "§7" + "'s horse";
+        String customNameSuffix = "§7" + " [" + playerName + "]";
+
+        Component horseName = horse.getCustomName();
+        if (horseName == null) return false;
+
+        return defaultName.equals(horseName.getString())
+                || horseName.getString().endsWith(customNameSuffix);
+    }
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/features/MountHorseHotkeyFeature.java:131]: this technically should be abstracted away into another folder, but this shouldn't matter for now

**Review by Incompleteusern — COMMENTED** [review]
a thing

**Review by Incompleteusern — DISMISSED** [review]

File: common/src/main/java/com/wynntils/features/MountHorseHotkeyFeature.java [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    private static final int searchRadius = 6; // Furthest blocks away from which we can interact with a horse
+    private static final int summonAttempts = 8;
+    private static final int summonDelayTicks = 5;
+    private static int prevItem = -1;
+    private static boolean alreadySetPrevItem = false;
+
+    public enum MountHorseStatus {
+        NO_HORSE("feature.wynntils.mountHorseHotkey.noHorse"),
+        ALREADY_RIDING("feature.wynntils.mountHorseHotkey.alreadyRiding"),
+        ;
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/MountHorseHotkeyFeature.java]: Can't you put `;` instead of the last `,`?

**Review by kristofbolyai — DISMISSED** [review]

File: common/src/main/java/com/wynntils/features/MountHorseHotkeyFeature.java (hunk 4 above)
Comment (ryanzhoudev) [reply common/src/main/java/com/wynntils/features/MountHorseHotkeyFeature.java]: Yea, guess I ignored it cause I thought spotless would clean that up

**Review by kristofbolyai — APPROVED** [review]

**Pull request merged**